src/anomalib/models/image/dinomaly/lightning_model.py

```diff
@@ -0,0 +1,471 @@
+        mask_neighbor_size (int): Size of neighborhood for attention masking in decoder.
+            Set to 0 to disable masking. Defaults to 0.
+        remove_class_token (bool): Whether to remove class token from features
+            before processing. Defaults to False.
+        encoder_require_grad_layer (list[int]): List of encoder layer indices
+            that require gradients during training. Empty list freezes all encoder
+            layers. Defaults to empty list.
+        pre_processor (PreProcessor | bool, optional): Pre-processor instance or
+            flag to use default. Defaults to ``True``.
+        post_processor (PostProcessor | bool, optional): Post-processor instance
+            or flag to use default. Defaults to ``True``.
+        evaluator (Evaluator | bool, optional): Evaluator instance or flag to use
+            default. Defaults to ``True``.
+        visualizer (Visualizer | bool, optional): Visualizer instance or flag to
+            use default. Defaults to ``True``.
+
+    Example:
+        >>> from anomalib.data import MVTecAD
+        >>> from anomalib.models import Dinomaly
+        >>>
+        >>> # Basic usage with default parameters
+        >>> model = Dinomaly()
+        >>>
+        >>> # Custom configuration
+        >>> model = Dinomaly(
+        ...     encoder_name="dinov2reg_vit_large_14",
+        ...     decoder_depth=12,
+        ...     bottleneck_dropout=0.1,
+        ...     mask_neighbor_size=3
+        ... )
+        >>>
+        >>> # Training with datamodule
+        >>> datamodule = MVTecAD()
+        >>> engine = Engine()
+        >>> engine.fit(model, datamodule=datamodule)
+
+    Note:
+        The model requires significant GPU memory due to the Vision Transformer
+        architecture. Consider using gradient checkpointing or smaller model
+        variants for memory-constrained environments.
+    """
+
+    def __init__(
+        self,
+        encoder_name: str = "dinov2reg_vit_base_14",
+        bottleneck_dropout: float = 0.2,
+        decoder_depth: int = 8,
+        target_layers: list[int] | None = None,
+        fuse_layer_encoder: list[list[int]] | None = None,
+        fuse_layer_decoder: list[list[int]] | None = None,
+        mask_neighbor_size: int = 0,
+        remove_class_token: bool = False,
+        encoder_require_grad_layer: list[int] | None = None,
+        pre_processor: PreProcessor | bool = True,
+        post_processor: PostProcessor | bool = True,
+        evaluator: Evaluator | bool = True,
+        visualizer: Visualizer | bool = True,
+    ) -> None:
+        super().__init__(
+            pre_processor=pre_processor,
+            post_processor=post_processor,
+            evaluator=evaluator,
+            visualizer=visualizer,
+        )
+        # Handle None case for encoder_require_grad_layer
+        if encoder_require_grad_layer is None:
+            encoder_require_grad_layer = []
+
+        self.model: ViTill = ViTill(
```

Comment:
Following anomalib's torch model naming convention, can we name this model as `DinomalyModel`?